instrumentation/pulsar/pulsar-2.8/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/PulsarClientTest.java

```diff
@@ -0,0 +1,657 @@
+  private static Consumer<String> consumer;
+  private static Producer<String> producer2;
+
+  private static String brokerHost;
+  private static int brokerPort;
+
+  private static final AttributeKey<String> MESSAGE_TYPE =
+      AttributeKey.stringKey("messaging.pulsar.message.type");
+
+  @BeforeAll
+  static void beforeAll() throws PulsarClientException {
+    pulsar =
+        new PulsarContainer(DEFAULT_IMAGE_NAME)
+            .withEnv("PULSAR_MEM", "-Xmx128m")
+            .withLogConsumer(new Slf4jLogConsumer(logger))
+            .withStartupTimeout(Duration.ofMinutes(2));
+    pulsar.start();
+
+    brokerHost = pulsar.getHost();
+    brokerPort = pulsar.getMappedPort(6650);
+    client = PulsarClient.builder().serviceUrl(pulsar.getPulsarBrokerUrl()).build();
+    admin = PulsarAdmin.builder().serviceHttpUrl(pulsar.getHttpServiceUrl()).build();
+  }
+
+  @AfterAll
+  static void afterAll() throws PulsarClientException {
+    if (producer != null) {
+      producer.close();
+    }
+    if (consumer != null) {
+      consumer.close();
+    }
+    if (producer2 != null) {
+      producer2.close();
+    }
+    if (client != null) {
+      client.close();
+    }
+    if (admin != null) {
+      admin.close();
+    }
+    pulsar.close();
+  }
+
+  @Test
+  void testSendNonPartitionedTopic() throws Exception {
+    String topic = "persistent://public/default/testSendNonPartitionedTopic";
+    admin.topics().createNonPartitionedTopic(topic);
+    producer = client.newProducer(Schema.STRING).topic(topic).enableBatching(false).create();
+
+    String msg = "test";
+    MessageId msgId = testing.runWithSpan("parent", () -> producer.send(msg));
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                span ->
+                    span.hasName(topic + " publish")
+                        .hasKind(SpanKind.PRODUCER)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            sendAttributes(topic, msgId.toString(), false))));
+  }
+
+  @Test
+  void testConsumeNonPartitionedTopic() throws Exception {
+    String topic = "persistent://public/default/testConsumeNonPartitionedTopic";
+    CountDownLatch latch = new CountDownLatch(1);
+    admin.topics().createNonPartitionedTopic(topic);
+    consumer =
+        client
+            .newConsumer(Schema.STRING)
+            .subscriptionName("test_sub")
+            .topic(topic)
+            .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+            .messageListener(
+                (MessageListener<String>)
+                    (consumer, msg) -> {
+                      try {
+                        consumer.acknowledge(msg);
+                      } catch (PulsarClientException e) {
+                        throw new AssertionError(e);
```

Comment:
It's not an error related to an assertion. The semantic of a previous code version with a `RuntimeException` seems more appropriate, or with an `IllegalStateException`. Same remark for other tests.